lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -504,10 +531,10 @@ def write(chunk)
             labels_utf8_pairs = labels.map do |k, v|
               [k.encode('utf-8'), convert_to_utf8(v)]
             end
-            utf8_log_name = convert_to_utf8(log_name)
 
+            log_name = ERB::Util.url_encode(log_name)
```

Comment:
Moved this into the `def log_name` method. Note that we can only encode the log name for the grpc path, since the non-grpc path already handles encoding.